Timezone issues in DB

These tests fail, due to what seem to be timezone issues (the hour is 5 hours ahead/behind, I'm in UTC-5). This probably doesn't show up if the system clock is in UTC.

```
============================================================================================= test session starts ==============================================================================================
platform darwin -- Python 3.8.6, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /Users/aapeli/git/couchers/app/backend/src, configfile: pytest.ini
plugins: pythonpath-0.7.3
collected 74 items
run-last-failure: no previously failed tests, not deselecting items.

src/tests/test_api.py ................                                                                                                                                                                   [ 21%]
src/tests/test_auth.py ......                                                                                                                                                                            [ 29%]
src/tests/test_bug_tool.py ...                                                                                                                                                                           [ 33%]
src/tests/test_conversations.py .......F....F............                                                                                                                                                [ 67%]
src/tests/test_db.py ....                                                                                                                                                                                [ 72%]
src/tests/test_email.py .......                                                                                                                                                                          [ 82%]
src/tests/test_media.py .                                                                                                                                                                                [ 83%]
src/tests/test_models.py ...                                                                                                                                                                             [ 87%]
src/tests/test_requests.py .........                                                                                                                                                                     [100%]

=================================================================================================== FAILURES ===================================================================================================
___________________________________________________________________________________________ test_get_group_chat_info ___________________________________________________________________________________________

db = sessionmaker(class_='Session', bind=Engine(postgresql://postgres:***@localhost:6544/postgres), autoflush=True, autocommit=False, expire_on_commit=True)

>   ???
E   assert datetime.datetime(2020, 11, 21, 18, 35, 24, 998222) <= datetime.datetime(2020, 11, 21, 13, 35, 25, 179729)
E    +  where datetime.datetime(2020, 11, 21, 18, 35, 24, 998222) = <bound method Timestamp.ToDatetime of seconds: 1605983724\nnanos: 998222000\n>()
E    +    where <bound method Timestamp.ToDatetime of seconds: 1605983724\nnanos: 998222000\n> = seconds: 1605983724\nnanos: 998222000\n.ToDatetime
E    +      where seconds: 1605983724\nnanos: 998222000\n = group_chat_id: 1\ntitle: "Test title"\nmember_user_ids: 2\nmember_user_ids: 1\nadmin_user_ids: 1\nonly_admins_invite: true\n...  author_user_id: 1\n  time {\n    seconds: 1605983725\n    nanos: 59876000\n  }\n  text {\n    text: "Test message 2"\n  }\n}\n.created
E    +  and   datetime.datetime(2020, 11, 21, 13, 35, 25, 179729) = <built-in method now of type object at 0x10c6da668>()
E    +    where <built-in method now of type object at 0x10c6da668> = datetime.now

/app/src/tests/test_conversations.py:395: AssertionError
______________________________________________________________________________________________ test_send_message _______________________________________________________________________________________________

db = sessionmaker(class_='Session', bind=Engine(postgresql://postgres:***@localhost:6544/postgres), autoflush=True, autocommit=False, expire_on_commit=True)

>   ???
E   assert datetime.datetime(2020, 11, 21, 18, 35, 30, 196730) <= datetime.datetime(2020, 11, 21, 13, 35, 30, 214964)
E    +  where datetime.datetime(2020, 11, 21, 18, 35, 30, 196730) = <bound method Timestamp.ToDatetime of seconds: 1605983730\nnanos: 196730000\n>()
E    +    where <bound method Timestamp.ToDatetime of seconds: 1605983730\nnanos: 196730000\n> = seconds: 1605983730\nnanos: 196730000\n.ToDatetime
E    +      where seconds: 1605983730\nnanos: 196730000\n = message_id: 2\nauthor_user_id: 1\ntime {\n  seconds: 1605983730\n  nanos: 196730000\n}\ntext {\n  text: "Test message 1"\n}\n.time
E    +  and   datetime.datetime(2020, 11, 21, 13, 35, 30, 214964) = <built-in method now of type object at 0x10c6da668>()
E    +    where <built-in method now of type object at 0x10c6da668> = datetime.now

/app/src/tests/test_conversations.py:584: AssertionError
=========================================================================================== short test summary info ============================================================================================
FAILED src/tests/test_conversations.py::test_get_group_chat_info - assert datetime.datetime(2020, 11, 21, 18, 35, 24, 998222) <= datetime.datetime(2020, 11, 21, 13, 35, 25, 179729)
FAILED src/tests/test_conversations.py::test_send_message - assert datetime.datetime(2020, 11, 21, 18, 35, 30, 196730) <= datetime.datetime(2020, 11, 21, 13, 35, 30, 214964)
=================================================================================== 2 failed, 72 passed in 66.76s (0:01:06) ====================================================================================
```

--- a/app/backend/src/couchers/models.py
+++ b/app/backend/src/couchers/models.py
@@ -63,6 +63,7 @@ class User(Base):
     phone = Column(String, nullable=True, unique=True)
     phone_status = Column(Enum(PhoneStatus), nullable=True)
 
+    # timezones should always be UTC
     joined = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
     last_active = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
 
@@ -70,7 +71,7 @@ class User(Base):
     name = Column(String, nullable=False)
     city = Column(String, nullable=False)
     gender = Column(String, nullable=False)
-    birthdate = Column(Date, nullable=False)
+    birthdate = Column(Date, nullable=False)  # in the timezone of birthplace
 
     # name as on official docs for verification, etc. not needed until verification
     full_name = Column(String, nullable=True)
@@ -203,6 +204,7 @@ class FriendRelationship(Base):
 
     status = Column(Enum(FriendStatus), nullable=False, default=FriendStatus.pending)
 
+    # timezones should always be UTC
     time_sent = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
     time_responded = Column(DateTime(timezone=True), nullable=True)
 
@@ -220,6 +222,7 @@ class SignupToken(Base):
 
     email = Column(String, nullable=False)
 
+    # timezones should always be UTC
     created = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
     expiry = Column(DateTime(timezone=True), nullable=False)
 
@@ -237,6 +240,7 @@ class LoginToken(Base):
 
     user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
 
+    # timezones should always be UTC
     created = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
     expiry = Column(DateTime(timezone=True), nullable=False)
 
@@ -256,6 +260,7 @@ class UserSession(Base):
 
     user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
 
+    # timezone should always be UTC
     started = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
 
     user = relationship("User", backref="sessions")
@@ -276,6 +281,7 @@ class Reference(Base):
     __table_args__ = (UniqueConstraint("from_user_id", "to_user_id", "reference_type"),)
 
     id = Column(Integer, primary_key=True)
+    # timezone should always be UTC
     time = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
 
     from_user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
@@ -300,6 +306,7 @@ class Conversation(Base):
     __tablename__ = "conversations"
 
     id = Column(Integer, primary_key=True)
+    # timezone should always be UTC
     created = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
 
     def __repr__(self):
@@ -344,6 +351,7 @@ class GroupChatSubscription(Base):
     user_id = Column(ForeignKey("users.id"), nullable=False)
     group_chat_id = Column(ForeignKey("group_chats.id"), nullable=False)
 
+    # timezones should always be UTC
     joined = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
     left = Column(DateTime(timezone=True), nullable=True)
 
@@ -418,7 +426,7 @@ class Message(Base):
     # the target if a control message and requires target, e.g. if inviting a user, the user invited is the target
     target_id = Column(ForeignKey("users.id"), nullable=True)
 
-    # time sent
+    # time sent, timezone should always be UTC
     time = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
 
     # the message text if not control
@@ -451,6 +459,7 @@ class Complaint(Base):
 
     id = Column(Integer, primary_key=True)
 
+    # timezone should always be UTC
     time = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
 
     author_user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
@@ -471,6 +480,8 @@ class Email(Base):
     __tablename__ = "emails"
 
     id = Column(String, primary_key=True)
+
+    # timezone should always be UTC
     time = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
 
     sender_name = Column(String, nullable=False)
@@ -494,7 +505,7 @@ class HostRequest(Base):
     from_user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
     to_user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
 
-    # dates as "YYYY-MM-DD"
+    # dates as "YYYY-MM-DD", in the timezone of the host
     from_date = Column(String, nullable=False)
     to_date = Column(String, nullable=False)
 
@@ -522,6 +533,7 @@ class InitiatedUpload(Base):
 
     key = Column(String, primary_key=True)
 
+    # timezones should always be UTC
     created = Column(DateTime(timezone=True), nullable=False, server_default=func.now())
     expiry = Column(DateTime(timezone=True), nullable=False)
 

--- a/app/backend/src/couchers/servicers/api.py
+++ b/app/backend/src/couchers/servicers/api.py
@@ -1,6 +1,6 @@
 import logging
 from base64 import urlsafe_b64decode, urlsafe_b64encode
-from datetime import datetime, timedelta
+from datetime import timedelta
 from urllib.parse import urlencode
 
 import grpc

--- a/app/backend/src/couchers/servicers/conversations.py
+++ b/app/backend/src/couchers/servicers/conversations.py
@@ -1,5 +1,4 @@
 import logging
-from datetime import datetime
 
 import grpc
 from google.protobuf import empty_pb2

--- a/app/backend/src/couchers/servicers/requests.py
+++ b/app/backend/src/couchers/servicers/requests.py
@@ -1,3 +1,4 @@
+import datetime
 import logging
 
 import grpc
@@ -10,7 +11,7 @@ from couchers import errors
 from couchers.db import is_valid_date, session_scope
 from couchers.models import Conversation, HostRequest, HostRequestStatus, Message, MessageType, User
 from couchers.tasks import send_host_request_email
-from couchers.utils import Timestamp_from_datetime, now
+from couchers.utils import Timestamp_from_datetime
 from pb import conversations_pb2, requests_pb2, requests_pb2_grpc
 
 logger = logging.getLogger(__name__)
@@ -54,6 +55,10 @@ def message_to_pb(message: Message):
         )
 
 
+def now_date_in_timezone(hours):
+    return datetime.datetime.now(datetime.timezone(datetime.timedelta(hours=hours))).strftime("%Y-%m-%d")
+
+
 class Requests(requests_pb2_grpc.RequestsServicer):
     def __init__(self, Session):
         self._Session = Session
@@ -70,10 +75,10 @@ class Requests(requests_pb2_grpc.RequestsServicer):
             if not is_valid_date(request.from_date) or not is_valid_date(request.to_date):
                 context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.INVALID_DATE)
 
-            if request.from_date > request.to_date:
+            if request.from_date >= request.to_date:
                 context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.DATE_FROM_AFTER_TO)
 
-            if request.to_date < now().strftime("%Y-%m-%d"):
+            if request.to_date < now_date_in_timezone(-12):
                 context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.DATE_TO_BEFORE_TODAY)
 
             conversation = Conversation()
@@ -193,8 +198,8 @@ class Requests(requests_pb2_grpc.RequestsServicer):
                         HostRequest.status == HostRequestStatus.confirmed,
                     )
                 )
-                # TODO: This uses the server timezone, how to use users'?
-                today = now().strftime("%Y-%m-%d")
+                # As we don't store the user's timezone yet, assume far east
+                today = now_date_in_timezone(+12)
                 # Is this string comparison a bad idea?
                 query = query.filter(HostRequest.to_date <= today)
 
@@ -244,8 +249,8 @@ class Requests(requests_pb2_grpc.RequestsServicer):
             if request.status == conversations_pb2.HOST_REQUEST_STATUS_PENDING:
                 context.abort(grpc.StatusCode.PERMISSION_DENIED, errors.INVALID_HOST_REQUEST_STATUS)
 
-            # TODO: user local time
-            today = now().strftime("%Y-%m-%d")
+            # As we don't store user's timezone yet, assume far west
+            today = now_date_in_timezone(-12)
             if host_request.to_date < today:
                 context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.HOST_REQUEST_IN_PAST)
 

--- a/app/backend/src/couchers/servicers/requests.py
+++ b/app/backend/src/couchers/servicers/requests.py
@@ -1,4 +1,3 @@
-import datetime
 import logging
 
 import grpc
@@ -11,7 +10,7 @@ from couchers import errors
 from couchers.db import is_valid_date, session_scope
 from couchers.models import Conversation, HostRequest, HostRequestStatus, Message, MessageType, User
 from couchers.tasks import send_host_request_email
-from couchers.utils import Timestamp_from_datetime
+from couchers.utils import Timestamp_from_datetime, now
 from pb import conversations_pb2, requests_pb2, requests_pb2_grpc
 
 logger = logging.getLogger(__name__)
@@ -74,7 +73,7 @@ class Requests(requests_pb2_grpc.RequestsServicer):
             if request.from_date > request.to_date:
                 context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.DATE_FROM_AFTER_TO)
 
-            if request.to_date < datetime.datetime.now().strftime("%Y-%m-%d"):
+            if request.to_date < now().strftime("%Y-%m-%d"):
                 context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.DATE_TO_BEFORE_TODAY)
 
             conversation = Conversation()
@@ -195,7 +194,7 @@ class Requests(requests_pb2_grpc.RequestsServicer):
                     )
                 )
                 # TODO: This uses the server timezone, how to use users'?
-                today = datetime.datetime.now().strftime("%Y-%m-%d")
+                today = now().strftime("%Y-%m-%d")
                 # Is this string comparison a bad idea?
                 query = query.filter(HostRequest.to_date <= today)
 
@@ -246,7 +245,7 @@ class Requests(requests_pb2_grpc.RequestsServicer):
                 context.abort(grpc.StatusCode.PERMISSION_DENIED, errors.INVALID_HOST_REQUEST_STATUS)
 
             # TODO: user local time
-            today = datetime.datetime.now().strftime("%Y-%m-%d")
+            today = now().strftime("%Y-%m-%d")
             if host_request.to_date < today:
                 context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.HOST_REQUEST_IN_PAST)
 

--- a/app/backend/src/tests/test_api.py
+++ b/app/backend/src/tests/test_api.py
@@ -1,4 +1,4 @@
-from datetime import datetime, timedelta
+from datetime import timedelta
 
 import grpc
 import pytest
@@ -6,7 +6,7 @@ from google.protobuf import empty_pb2, wrappers_pb2
 
 from couchers.db import session_scope
 from couchers.models import Complaint
-from couchers.utils import to_aware_datetime
+from couchers.utils import now, to_aware_datetime
 from pb import api_pb2
 from tests.test_fixtures import api_session, db, generate_user, make_friends, real_api_session, testconfig
 
@@ -514,7 +514,7 @@ def test_references(db):
         assert res.references[0].from_user_id == user1.id
         assert res.references[0].to_user_id == user2.id
         assert res.references[0].text == "kinda weird sometimes"
-        assert abs(res.references[0].written_time.ToDatetime() - datetime.now()) <= timedelta(days=32)
+        assert abs(to_aware_datetime(res.references[0].written_time) - now()) <= timedelta(days=32)
         assert res.references[0].reference_type not in seen_types
         seen_types.add(res.references[0].reference_type)
     assert seen_types == alltypes

--- a/app/backend/src/tests/test_conversations.py
+++ b/app/backend/src/tests/test_conversations.py
@@ -365,6 +365,7 @@ def test_get_group_chat_messages_joined_left(db):
         assert res.messages[5].WhichOneof("content") == "user_invited"
 
 
+@pytest.mark.xfail
 def test_get_group_chat_info(db):
     user1, token1 = generate_user(db)
     user2, token2 = generate_user(db)
@@ -568,6 +569,7 @@ def test_make_remove_group_chat_admin(db):
         assert e.value.code() == grpc.StatusCode.PERMISSION_DENIED
 
 
+@pytest.mark.xfail
 def test_send_message(db):
     user1, token1 = generate_user(db)
     user2, token2 = generate_user(db)

--- a/app/backend/src/tests/test_conversations.py
+++ b/app/backend/src/tests/test_conversations.py
@@ -1,11 +1,11 @@
-from datetime import datetime
-
 import grpc
 import pytest
+import pytz
 from google.protobuf import empty_pb2, wrappers_pb2
 
 from couchers import errors
 from couchers.models import User
+from couchers.utils import now, to_aware_datetime
 from pb import api_pb2, conversations_pb2
 from tests.test_fixtures import api_session, conversations_session, db, generate_user, make_friends, testconfig
 
@@ -316,7 +316,7 @@ def test_get_group_chat_messages_joined_left(db):
     make_friends(db, user1, user2)
     make_friends(db, user1, user3)
     make_friends(db, user1, user4)
-    start_time = datetime.now()
+    start_time = now()
 
     with conversations_session(db, token1) as c:
         res = c.CreateGroupChat(conversations_pb2.CreateGroupChatReq(recipient_user_ids=[user2.id, user4.id]))
@@ -365,7 +365,6 @@ def test_get_group_chat_messages_joined_left(db):
         assert res.messages[5].WhichOneof("content") == "user_invited"
 
 
-@pytest.mark.xfail
 def test_get_group_chat_info(db):
     user1, token1 = generate_user(db)
     user2, token2 = generate_user(db)
@@ -393,7 +392,7 @@ def test_get_group_chat_info(db):
         assert res.title == "Test title"
         assert user2.id in res.member_user_ids
         assert user1.id in res.admin_user_ids
-        assert res.created.ToDatetime() <= datetime.now()
+        assert to_aware_datetime(res.created) <= now()
         assert res.only_admins_invite
         assert res.is_dm
 
@@ -402,7 +401,7 @@ def test_get_group_chat_info(db):
         assert user2.id in res.member_user_ids
         assert user3.id in res.member_user_ids
         assert user1.id in res.admin_user_ids
-        assert res.created.ToDatetime() <= datetime.now()
+        assert to_aware_datetime(res.created) <= now()
         assert res.only_admins_invite
         assert not res.is_dm
 
@@ -569,7 +568,6 @@ def test_make_remove_group_chat_admin(db):
         assert e.value.code() == grpc.StatusCode.PERMISSION_DENIED
 
 
-@pytest.mark.xfail
 def test_send_message(db):
     user1, token1 = generate_user(db)
     user2, token2 = generate_user(db)
@@ -583,7 +581,7 @@ def test_send_message(db):
         c.SendMessage(conversations_pb2.SendMessageReq(group_chat_id=group_chat_id, text="Test message 1"))
         res = c.GetGroupChatMessages(conversations_pb2.GetGroupChatMessagesReq(group_chat_id=group_chat_id))
         assert res.messages[0].text.text == "Test message 1"
-        assert res.messages[0].time.ToDatetime() <= datetime.now()
+        assert to_aware_datetime(res.messages[0].time) <= now()
         assert res.messages[0].author_user_id == user1.id
 
     # can't send message if not in chat
@@ -789,7 +787,7 @@ def test_search_messages_left_joined(db):
     make_friends(db, user1, user2)
     make_friends(db, user1, user3)
     make_friends(db, user1, user4)
-    start_time = datetime.now()
+    start_time = now()
 
     with conversations_session(db, token1) as c:
         res = c.CreateGroupChat(conversations_pb2.CreateGroupChatReq(recipient_user_ids=[user2.id, user4.id]))
@@ -1095,7 +1093,7 @@ def test_total_unseen(db):
     # distractions
     make_friends(db, user1, user4)
 
-    start_time = datetime.utcnow()
+    start_time = now()
 
     with conversations_session(db, token1) as c:
         # distractions

--- a/app/backend/src/tests/test_fixtures.py
+++ b/app/backend/src/tests/test_fixtures.py
@@ -1,3 +1,4 @@
+import os
 from concurrent import futures
 from contextlib import contextmanager
 from datetime import date, timedelta
@@ -36,6 +37,11 @@ def db():
     """
     Connect to a running Postgres database, and return the Session object.
     """
+
+    # running in non-UTC catches some timezone errors
+    # os.environ["TZ"] = "Etc/UTC"
+    os.environ["TZ"] = "America/New_York"
+
     engine = create_engine(config["DATABASE_CONNECTION_STRING"], poolclass=NullPool)
 
     Base.metadata.drop_all(engine)

--- a/app/backend/src/tests/test_requests.py
+++ b/app/backend/src/tests/test_requests.py
@@ -1,4 +1,4 @@
-from datetime import datetime, timedelta
+from datetime import timedelta
 
 import grpc
 import pytest
@@ -9,6 +9,7 @@ from sqlalchemy.sql import and_
 from couchers import errors
 from couchers.db import session_scope
 from couchers.models import Conversation, HostRequest, HostRequestStatus, Message, MessageType
+from couchers.utils import now
 from pb import api_pb2, conversations_pb2, requests_pb2
 from tests.test_fixtures import api_session, db, generate_user, make_friends, requests_session, testconfig
 
@@ -21,10 +22,10 @@ def _(testconfig):
 def test_create_request(db):
     user1, token1 = generate_user(db)
     user2, token2 = generate_user(db)
-    today_plus_1 = (datetime.now() + timedelta(days=1)).strftime("%Y-%m-%d")
-    today_plus_2 = (datetime.now() + timedelta(days=2)).strftime("%Y-%m-%d")
-    today_minus_1 = (datetime.now() - timedelta(days=1)).strftime("%Y-%m-%d")
-    today_minus_2 = (datetime.now() - timedelta(days=2)).strftime("%Y-%m-%d")
+    today_plus_1 = (now() + timedelta(days=1)).strftime("%Y-%m-%d")
+    today_plus_2 = (now() + timedelta(days=2)).strftime("%Y-%m-%d")
+    today_minus_1 = (now() - timedelta(days=1)).strftime("%Y-%m-%d")
+    today_minus_2 = (now() - timedelta(days=2)).strftime("%Y-%m-%d")
     with requests_session(db, token1) as api:
         with pytest.raises(grpc.RpcError) as e:
             api.CreateHostRequest(
@@ -97,8 +98,8 @@ def test_get_host_request(db):
     user1, token1 = generate_user(db)
     user2, token2 = generate_user(db)
     user3, token3 = generate_user(db)
-    today_plus_1 = (datetime.now() + timedelta(days=1)).strftime("%Y-%m-%d")
-    today_plus_2 = (datetime.now() + timedelta(days=2)).strftime("%Y-%m-%d")
+    today_plus_1 = (now() + timedelta(days=1)).strftime("%Y-%m-%d")
+    today_plus_2 = (now() + timedelta(days=2)).strftime("%Y-%m-%d")
     with requests_session(db, token1) as api:
         host_request_id = api.CreateHostRequest(
             requests_pb2.CreateHostRequestReq(
@@ -122,8 +123,8 @@ def test_list_requests(db):
     user1, token1 = generate_user(db)
     user2, token2 = generate_user(db)
     user3, token3 = generate_user(db)
-    today_plus_1 = (datetime.now() + timedelta(days=1)).strftime("%Y-%m-%d")
-    today_plus_2 = (datetime.now() + timedelta(days=2)).strftime("%Y-%m-%d")
+    today_plus_1 = (now() + timedelta(days=1)).strftime("%Y-%m-%d")
+    today_plus_2 = (now() + timedelta(days=2)).strftime("%Y-%m-%d")
     with requests_session(db, token1) as api:
         host_request_1 = api.CreateHostRequest(
             requests_pb2.CreateHostRequestReq(
@@ -183,8 +184,8 @@ def test_list_requests(db):
 def test_list_host_requests_active_filter(db):
     user1, token1 = generate_user(db)
     user2, token2 = generate_user(db)
-    today_plus_1 = (datetime.now() + timedelta(days=1)).strftime("%Y-%m-%d")
-    today_plus_2 = (datetime.now() + timedelta(days=2)).strftime("%Y-%m-%d")
+    today_plus_1 = (now() + timedelta(days=1)).strftime("%Y-%m-%d")
+    today_plus_2 = (now() + timedelta(days=2)).strftime("%Y-%m-%d")
 
     with requests_session(db, token1) as api:
         request_id = api.CreateHostRequest(
@@ -209,8 +210,8 @@ def test_respond_host_requests(db):
     user1, token1 = generate_user(db)
     user2, token2 = generate_user(db)
     user3, token3 = generate_user(db)
-    today_plus_1 = (datetime.now() + timedelta(days=1)).strftime("%Y-%m-%d")
-    today_plus_2 = (datetime.now() + timedelta(days=2)).strftime("%Y-%m-%d")
+    today_plus_1 = (now() + timedelta(days=1)).strftime("%Y-%m-%d")
+    today_plus_2 = (now() + timedelta(days=2)).strftime("%Y-%m-%d")
 
     with requests_session(db, token1) as api:
         request_id = api.CreateHostRequest(
@@ -320,8 +321,8 @@ def test_respond_host_requests(db):
 def test_get_host_request_messages(db):
     user1, token1 = generate_user(db)
     user2, token2 = generate_user(db)
-    today_plus_1 = (datetime.now() + timedelta(days=1)).strftime("%Y-%m-%d")
-    today_plus_2 = (datetime.now() + timedelta(days=2)).strftime("%Y-%m-%d")
+    today_plus_1 = (now() + timedelta(days=1)).strftime("%Y-%m-%d")
+    today_plus_2 = (now() + timedelta(days=2)).strftime("%Y-%m-%d")
     with requests_session(db, token1) as api:
         res = api.CreateHostRequest(
             requests_pb2.CreateHostRequestReq(
@@ -386,8 +387,8 @@ def test_send_message(db):
     user1, token1 = generate_user(db)
     user2, token2 = generate_user(db)
     user3, token3 = generate_user(db)
-    today_plus_1 = (datetime.now() + timedelta(days=1)).strftime("%Y-%m-%d")
-    today_plus_2 = (datetime.now() + timedelta(days=2)).strftime("%Y-%m-%d")
+    today_plus_1 = (now() + timedelta(days=1)).strftime("%Y-%m-%d")
+    today_plus_2 = (now() + timedelta(days=2)).strftime("%Y-%m-%d")
     with requests_session(db, token1) as api:
         host_request_id = api.CreateHostRequest(
             requests_pb2.CreateHostRequestReq(
@@ -479,8 +480,8 @@ def test_get_updates(db):
     user1, token1 = generate_user(db)
     user2, token2 = generate_user(db)
     user3, token3 = generate_user(db)
-    today_plus_1 = (datetime.now() + timedelta(days=1)).strftime("%Y-%m-%d")
-    today_plus_2 = (datetime.now() + timedelta(days=2)).strftime("%Y-%m-%d")
+    today_plus_1 = (now() + timedelta(days=1)).strftime("%Y-%m-%d")
+    today_plus_2 = (now() + timedelta(days=2)).strftime("%Y-%m-%d")
     with requests_session(db, token1) as api:
         host_request_id = api.CreateHostRequest(
             requests_pb2.CreateHostRequestReq(
@@ -554,8 +555,8 @@ def test_mark_last_seen(db):
     user1, token1 = generate_user(db)
     user2, token2 = generate_user(db)
     user3, token3 = generate_user(db)
-    today_plus_1 = (datetime.now() + timedelta(days=1)).strftime("%Y-%m-%d")
-    today_plus_2 = (datetime.now() + timedelta(days=2)).strftime("%Y-%m-%d")
+    today_plus_1 = (now() + timedelta(days=1)).strftime("%Y-%m-%d")
+    today_plus_2 = (now() + timedelta(days=2)).strftime("%Y-%m-%d")
     with requests_session(db, token1) as api:
         host_request_id = api.CreateHostRequest(
             requests_pb2.CreateHostRequestReq(

--- a/app/pb/auth.proto
+++ b/app/pb/auth.proto
@@ -98,7 +98,7 @@ message CompleteSignupReq {
   string username = 2;
   string name = 3;
   string city = 4;
-  string birthdate = 5; // in YYYY-MM-DD format
+  string birthdate = 5; // in YYYY-MM-DD format, in the timezone of the birth place
   string gender = 6;
   org.couchers.api.core.HostingStatus hosting_status = 7;
 }

--- a/app/pb/requests.proto
+++ b/app/pb/requests.proto
@@ -39,7 +39,7 @@ service Requests {
 
 message CreateHostRequestReq {
   uint64 to_user_id = 1;
-  // dates as "yyyy-mm-dd"
+  // dates as "yyyy-mm-dd", in the timezone of the host
   string from_date = 2;
   string to_date = 3;
   string text = 4;
@@ -51,6 +51,8 @@ message HostRequest {
   uint64 to_user_id = 3;
   org.couchers.api.conversations.HostRequestStatus status = 4;
   google.protobuf.Timestamp created = 5;
+
+  // dates as "yyyy-mm-dd", in the timezone of the host
   string from_date = 6;
   string to_date = 7;
   uint64 last_seen_message_id = 8;

--- a/docs/contributing.md
+++ b/docs/contributing.md
@@ -69,5 +69,3 @@ Additionally, we strive to use the ["Google" docstring format](https://sphinxcon
 Please use double quotes `"`, instead of single quotes `'`.
 
 In the **frontend**, make sure to run `yarn format` in `app/frontend` before asking for a review.
-
-The server should operate in UTC, and we should always use datetimes with timezones. Please use the utility functions in `couchers.utils`. For now we pretend the whole world operates in UTC, which will of course have to change soon given we're going to care about local times. How this responsibility is divided up between the frontend and backend is not entirely clear yet though.

--- a/docs/contributing.md
+++ b/docs/contributing.md
@@ -68,4 +68,6 @@ Additionally, we strive to use the ["Google" docstring format](https://sphinxcon
 
 Please use double quotes `"`, instead of single quotes `'`.
 
-In the **frontend**, make sure to run `yarn format` in app/frontend before asking for a review.
+In the **frontend**, make sure to run `yarn format` in `app/frontend` before asking for a review.
+
+The server should operate in UTC, and we should always use datetimes with timezones. Please use the utility functions in `couchers.utils`. For now we pretend the whole world operates in UTC, which will of course have to change soon given we're going to care about local times. How this responsibility is divided up between the frontend and backend is not entirely clear yet though.

